### PR TITLE
[CI-3691] - Support labels in results

### PR DESCRIPTION
**Pull request by stanlp1**
Less changes than there may seem. Majority of changes are formatting changes.

### CODE REVIEW

File: src/constructor.io/models/common/Result.cs [hunk 1]
```diff
@@ -26,5 +26,9 @@ public class Result
 
         [JsonProperty("strategy")]
         public RecommendationStrategy Strategy { get; set; }
+
+        [JsonProperty("labels")]
+        public Dictionary<string, object> Labels { get; set; }
```
Comment (stanlp1) [inline src/constructor.io/models/common/Result.cs:31]: Main Change

File: src/Constructorio_NET.Tests/client/modules/SearchTests.cs [hunk 2]
```diff
@@ -26,37 +26,51 @@ public void Setup()
...
             ConstructorIO constructorio = new ConstructorIO(this.Config);
             SearchResponse res = await constructorio.Search.GetSearchResults(req);
-            Assert.Greater(res.Response.TotalNumResults, 0, "total number of results expected to be greater than 0");
-            Assert.Greater(res.Response.Results.Count, 0, "length of results expected to be greater than 0");
-            Assert.Greater(res.Response.Facets.Count, 0, "length of facets expected to be greater than 0");
+            Dictionary<string, object> labels = res.Response.Results[0].Labels;
+
+            labels.TryGetValue("is_sponsored", out object isSponsored);
+
+            Assert.AreEqual((bool)isSponsored, true);
```
Comment (stanlp1) [inline src/Constructorio_NET.Tests/client/modules/SearchTests.cs:51]: Search Test Addition

File: src/Constructorio_NET.Tests/client/modules/AutocompleteTests.cs [hunk 3]
```diff
@@ -25,25 +26,30 @@ public void Setup()
...
-            };
+            AutocompleteRequest req = new AutocompleteRequest("item") { UserInfo = UserInfo };
             ConstructorIO constructorio = new ConstructorIO(this.Config);
             AutocompleteResponse res = await constructorio.Autocomplete.GetAutocompleteResults(req);
+            res.Sections.TryGetValue("Products", out List<Result> Products);
+            Dictionary<string, object> labels = Products[0].Labels;
+
+            labels.TryGetValue("is_sponsored", out object isSponsored);
 
+            Assert.AreEqual((bool)isSponsored, true);
```
Comment (stanlp1) [inline src/Constructorio_NET.Tests/client/modules/AutocompleteTests.cs:52]: Autocomplete test addition

File: src/Constructorio_NET.Tests/client/modules/BrowseTests.cs [hunk 4]
```diff
@@ -48,9 +51,26 @@ public async Task GetBrowseResults()
...
             ConstructorIO constructorio = new ConstructorIO(this.Config);
             BrowseResponse res = await constructorio.Browse.GetBrowseResults(req);
-            Assert.Greater(res.Response.TotalNumResults, 0, "total number of results expected to be greater than 0");
-            Assert.Greater(res.Response.Results.Count, 0, "length of results expected to be greater than 0");
-            Assert.Greater(res.Response.Facets.Count, 0, "length of facets expected to be greater than 0");
+            Dictionary<string, object> labels = res.Response.Results[0].Labels;
+
+            labels.TryGetValue("is_sponsored", out object isSponsored);
+
+            Assert.AreEqual((bool)isSponsored, true);
```
Comment (stanlp1) [inline src/Constructorio_NET.Tests/client/modules/BrowseTests.cs:58]: Browse Test addition

**Comment by mocca102** [comment]
Are linting changes necessary? Are the changes mainly what you added comments for? @stanlp1

**Comment by stanlp1** [comment]
> Are linting changes necessary? Are the changes mainly what you added comments for? @stanlp1

Not necessary but they just happened when I saved. I can see if I can disable and remove them.

And yes, the comments are basically the only changes. This was a simple PR.

**Review by crgee1 — APPROVED** [review]
lgtm

**Pull request merged**